### PR TITLE
Migrated http_redirect_test from SDK.

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

File: test/http_redirect_test.dart [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+    // Setup redirect where we close the connection.
+    addRequestHandler("/closing", (HttpRequest request, HttpResponse response) {
+      response.headers
+          .set(HttpHeaders.LOCATION, "http://127.0.0.1:${server.port}/");
+      response.statusCode = HttpStatus.FOUND;
+      response.persistentConnection = false;
+      response.close();
+    });
+
+    completer.complete(server);
```
Comment (zanderso) [inline test/http_redirect_test.dart]: return server;

File: test/http_redirect_test.dart [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+// BSD-style license that can be found in the LICENSE file.
+
+import "dart:async";
+
+import 'package:http_io/http_io.dart';
+import 'package:test/test.dart';
+
+Future<HttpServer> setupServer() {
+  Completer completer = new Completer();
+  HttpServer.bind("127.0.0.1", 0).then((server) {
```
Comment (zanderso) [inline test/http_redirect_test.dart]: return Http.bind(...).then((server) {...

File: test/http_redirect_test.dart [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import "dart:async";
+
+import 'package:http_io/http_io.dart';
+import 'package:test/test.dart';
+
+Future<HttpServer> setupServer() {
+  Completer completer = new Completer();
```
Comment (zanderso) [inline test/http_redirect_test.dart]: No need for a completer here, just return the server directly.

File: test/http_redirect_test.dart [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+      response.headers
+          .set(HttpHeaders.LOCATION, "http://127.0.0.1:${server.port}/");
+      response.statusCode = HttpStatus.FOUND;
+      response.persistentConnection = false;
+      response.close();
+    });
+
+    completer.complete(server);
+  });
+  return completer.future;
```
Comment (zanderso) [inline test/http_redirect_test.dart]: rm

File: test/http_redirect_test.dart [hunk 5]
```diff
@@ -0,0 +1,472 @@
...
+          client.close();
+          completer.complete();
+        });
+      });
+    });
+    return completer.future;
+  }
+
+  final futures = <Future<Null>>[];
+  futures.add(testPath("/redirectUrl"));
```
Comment (zanderso) [inline test/http_redirect_test.dart]: I'd prefer testing these more sequentially:
await testPath(...);
await testPath(...);

etc.

File: test/http_redirect_test.dart (hunk 1 above)
Comment (bkonyi) [reply test/http_redirect_test.dart]: Done

File: test/http_redirect_test.dart (hunk 2 above)
Comment (bkonyi) [reply test/http_redirect_test.dart]: Done

File: test/http_redirect_test.dart (hunk 3 above)
Comment (bkonyi) [reply test/http_redirect_test.dart]: Done

File: test/http_redirect_test.dart (hunk 4 above)
Comment (bkonyi) [reply test/http_redirect_test.dart]: Done

File: test/http_redirect_test.dart (hunk 5 above)
Comment (bkonyi) [reply test/http_redirect_test.dart]: Done

**Review by zanderso — APPROVED** [review]

**Pull request merged**